### ISSUE TITLE
[FEAT][MOE] fused_moe Op: end-to-end MoE pipeline, priority Qwen3 MoE

### ISSUE THREAD

**Issue by michaelwithu** [tile-ai/TileOPs#591]
## Parent Issue
Part of #404

## Overview
Wire all existing MoE sub-kernels into a single `Qwen3MoEOp` forward pass.
This is the first end-to-end integration target. fused_topk (#588) is **not** a prerequisite — use `torch.topk` + softmax as a drop-in for now.

## Pipeline

```
hidden_states [T, H]  +  gating_output [T, E]
    ↓ torch.topk + softmax (placeholder for #588)
topk_weights [T, K],  topk_ids [T, K]
    ↓ MoePermuteOp (#589, merged)
permuted_hidden [T*K, H],  expert_first_token_offset [E+1],  inv_permuted_idx [T*K]
    ↓ GroupedGemmOp (w1: [E, 2*inter, H])  ← adapt expert_first_token_offset → batch_sizes/offsets
mm1_out [T*K, 2*inter]
    ↓ SwiGLU: silu(gate) * up  (elementwise, in-place)
act_out [T*K, inter]
    ↓ GroupedGemmOp (w2: [E, H, inter])
mm2_out [T*K, H]
    ↓ MoeUnpermuteOp (#590, merged)
output [T, H]
```

## Scope (this PR)

- [ ] `tileops/ops/moe/qwen3_moe.py` — `Qwen3MoEOp` class
- [ ] Adapter: `expert_first_token_offset [E+1]` → `batch_sizes / batch_offsets` for GroupedGemmOp
- [ ] SwiGLU activation (torch or elementwise kernel, in-place on mm1_out)
- [ ] Correctness test vs decomposed PyTorch reference (Qwen3 MoE config: E=128, K=8, H=2048, inter=768)
- [ ] **Not in scope**: fused_topk (#588), fp8, tensor parallelism

## Interface

```python
class Qwen3MoEOp:
    def __init__(
        self,
        num_tokens: int,
        num_experts: int,       # 128
        top_k: int,             # 8
        hidden_size: int,       # 2048
        intermediate_size: int, # 768
        dtype: torch.dtype,     # bf16
    ): ...

    def forward(
        self,
        hidden_states: torch.Tensor,  # [T, H]
        gating_output: torch.Tensor,  # [T, E]
        w1: torch.Tensor,             # [E, 2*inter, H]
        w2: torch.Tensor,             # [E, H, inter]
    ) -> torch.Tensor: ...            # [T, H]
```

## Qwen3 MoE Config
| Parameter | Value |
|-----------|-------|
| num_experts | 128 |
| top_k | 8 |
| hidden_size | 2048 |
| intermediate_size | 768 |
| scoring_func | softmax |
| renormalize | False |
| activation | SiLU (SwiGLU) |
| shared experts | not included here |

## Sub-kernel Dependencies
- #589 MoePermuteOp ✅ merged
- #590 MoeUnpermuteOp ✅ merged
- GroupedGemmOp ✅ existing
